M2/Macaulay2/m2/matrix1.m2

```diff
@@ -485,18 +487,34 @@ generator Module := RingElement => (M) -> (
      if n == 1 then return M_0;
      error "expected ideal to have a single generator")
 
-Ideal / Ideal := Module => (I,J) -> module I / module J
+LeftIdeal / LeftIdeal := Module => (I,J) -> module I / module J
+
+-- two-sided ideal expected
 Module / Ideal := Module => (M,J) -> M / (J * M)
 
-Ideal#AfterPrint = Ideal#AfterNoPrint = (I) ->  (Ideal," of ",ring I)
+LeftIdeal#AfterPrint = LeftIdeal#AfterNoPrint = I ->  (LeftIdeal," of ",ring I)
+Ideal#AfterPrint = Ideal#AfterNoPrint = I ->  (Ideal," of ",ring I)
 
+-- two-sided ideal expected
 Ideal ^ ZZ := Ideal => (I,n) -> ideal symmetricPower(n,generators I)
 Ideal * Ideal := Ideal => ((I,J) -> ideal flatten (generators I ** generators J)) @@ samering
 Ideal * Module := Module => ((I,M) -> subquotient (generators I ** generators M, relations M)) @@ samering
-Ideal + Ideal := Ideal => ((I,J) -> ideal (generators I | generators J)) @@ tosamering
-Ideal + RingElement := Ideal + Number := ((I,r) -> I + ideal r) @@ tosamering
-RingElement + Ideal := Number + Ideal := ((r,I) -> ideal r + I) @@ tosamering
-Ideal _ ZZ := RingElement => (I,n) -> (generators I)_(0,n)
+
+LeftIdeal + LeftIdeal := LeftIdeal => ((I,J) -> ideal (generators I | generators J)) @@ tosamering
+-- Ideal + Ideal := Ideal => ((I,J) -> ideal (generators I | generators J)) @@ tosamering
+LeftIdeal + RingElement := LeftIdeal + Number := ((I,r) -> I + ideal r) @@ tosamering
+RingElement + LeftIdeal := Number + LeftIdeal := ((r,I) -> ideal r + I) @@ tosamering
+LeftIdeal _ ZZ := RingElement => (I,n) -> (generators I)_(0,n)
+
+
+Matrix % LeftIdeal := Matrix => ((f,I) -> 
+     if numRows f === 1
+     then f % gb I
+     else 
+     error "not implemented (defined for two-sided ideals; see `code (symbol %, Matrix, Ideal)`)"
```

Comment:
There is a comment in the two-sided code that says:
> we should have an engine routine to reduce each entry of a matrix modulo an ideal, to make this faster

If `RingElement % LeftIdeal` is defined, why not just run that on every element?